lucene/core/src/java/org/apache/lucene/util/fst/FST.java

```diff
@@ -864,6 +1012,68 @@ private long readUnpackedNodeTarget(BytesReader in) throws IOException {
     return readNextRealArc(arc, in);
   }
 
+  /**
+   * Reads the presence bits of a direct-addressing node, store them in the provided arc {@link Arc#bitTable()} and returns the number of presence bytes.
+   */
+  private int readPresenceBytes(Arc<T> arc, BytesReader in) throws IOException {
+    int numPresenceBytes = getNumPresenceBytes(arc.numArcs());
+    long[] presenceBits = new long[numPresenceBytes + 7 >>> 3];
+    int longIndex = 0;
+    int byteIndex = 0;
+    for (int i = 0; i < numPresenceBytes; i++) {
+      presenceBits[longIndex] |= (long) (in.readByte() & 0xFF) << (i << 3);
```

Comment:
```suggestion
      presenceBits[longIndex] |= (in.readByte() & 0xFFL) << (i << 3);
```

Reply:
maybe we should even remove longIndex/byteIndex for clarity and do

```
presenceBits[i >>>3] |= (in.readByte() & 0xFFL) << (i << 3);
```

Reply:
What about
presenceBits[i / Byte.SIZE] |= (in.readByte() & 0xFFL) << (i * Byte.SIZE);